lib/rules/no-early-return.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * @author *****your name*****
+ * See LICENSE file in root directory for full license.
+ */
+'use strict'
+
+// ------------------------------------------------------------------------------
+// Requirements
+// ------------------------------------------------------------------------------
+
+const utils = require('../utils')
+
+const PROPERTIES_NEED_CHECKING = new Set(['data', 'asyncData'])
+
+// ------------------------------------------------------------------------------
+// Rule Definition
+// ------------------------------------------------------------------------------
+
+module.exports = {
+  meta: {
+    type: 'problem',
+    docs: {
+      description: 'disallow early `returns` in `setup` and `data` functions',
+      categories: undefined,
+      url: 'https://eslint.vuejs.org/rules/no-early-return.html'
+    },
+    fixable: null,
+    schema: [],
+    messages: {
+      extraReturnStatement:
+        'Only one (unconditional) return statement is allowed in {{ functionType }} function.'
+    }
+  },
+
+  /** @param {RuleContext} context */
+  create(context) {
+    const returnStatementLocations = []
+
+    return utils.defineVueVisitor(context, {
+      ReturnStatement(node) {
+        if (context.getScope().type === 'function') {
+          return
+        }
+        returnStatementLocations.push(node.loc)
+      },
+
+      onSetupFunctionExit() {
```

Comment:
Can't the logic for `data`/`asyncData` and for `setup` be shared? I think they should work/behave exactly the same.